content/courses/program-security/closing-accounts.md

```diff
@@ -1,123 +1,53 @@
-  - Close program accounts safely and securely using native Rust
-  - Close program accounts safely and securely using the Anchor `close`
-    constraint
+  - Understand the security risks associated with improper account closing
+  - Implement secure account closing techniques in native Rust and Anchor
+  - Recognize and prevent revival attacks
+  - Apply best practices for account management in real-world Solana programs
 description:
-  "How to close program accounts safely and securely in Anchor and native Rust."
+  "Master the art of securely closing accounts in Solana programs to prevent
+  revival attacks and maintain program integrity."
 ---
 
 ## Summary
 
-- **Closing an account** improperly creates an opportunity for
-  reinitialization/revival attacks
-- The Solana runtime **garbage collects accounts** when they are no longer rent
-  exempt. Closing accounts involves transferring the lamports stored in the
-  account for rent exemption to another account of your choosing.
-- You can use the Anchor `#[account(close = <address_to_send_lamports>)]`
-  constraint to securely close accounts and set the account discriminator to the
-  `CLOSED_ACCOUNT_DISCRIMINATOR`
-  ```rust
-  #[account(mut, close = receiver)]
-  pub data_account: Account<'info, MyData>,
-  #[account(mut)]
-  pub receiver: SystemAccount<'info>
-  ```
+- Improper account closing can lead to revival attacks, compromising program
+  security
+- The Solana runtime garbage collects accounts when they are no longer rent
+  exempt
+- Securely close accounts by transferring lamports, zeroing out data, and
+  setting a closed account discriminator
+- Utilize Anchor's `close` constraint for simplified and secure account closing
+- Implement additional safeguards like force defund instructions to handle edge
+  cases
 
 ## Lesson
 
-While it sounds simple, closing accounts properly can be tricky. There are a
-number of ways an attacker could circumvent having the account closed if you
-don't follow specific steps.
+### Understanding Account Closing and Revival Attacks
 
-To get a better understanding of these attack vectors, let’s explore each of
-these scenarios in depth.
+Closing accounts properly is crucial for maintaining the security and integrity
+of your Solana program. Revival attacks occur when an attacker exploits
+improperly closed accounts to regain control or access unauthorized funds.
 
-### Insecure account closing
+### Real-World Implications: The Wormhole Hack
 
-At its core, closing an account involves transferring its lamports to a separate
-account, thus triggering the Solana runtime to garbage collect the first
-account. This resets the owner from the owning program to the system program.
+The Wormhole bridge hack, which resulted in the loss of 120,000 wETH,
+demonstrated the severe consequences of improper account management. While not
+directly related to account closing, it highlights the importance of robust
+account handling in DeFi protocols.
```

Comment:
Good content, but probably better in the section on signature verification (which was the basis of the wormhole attack).